[INTERNAL] Add preliminary guide for v3 migration (#608)
SAP/ui5-tooling: docs/updates/migrate-v3.md

# Migrate to v3

!!! warning
    UI5 Tooling v3 is currently in development. Further breaking changes are expected.  
    The latest development version can be installed via `npm i -D @ui5/cli@next`

## Breaking changes

> **All UI5 Tooling Modules: Require Node.js >= 16.13.2 / npm >= 8**

Reviewer: ```suggestion
**All UI5 Tooling Modules Require Node.js >= 16.13.2 / npm >= 8**
```

Support for older Node.js and npm releases has been dropped.
Only Node.js v16.13.2 and npm v8 or higher are supported.

**Removal of tasks and processors**

The following tasks have been removed:

- createDebugFiles
- uglify

The following processors have been removed:

- debugFileCreator
- resourceCopier
- uglifier

As a replacement, the new 'minify' task and 'minifier' processor can be
used.

Note: The minify task is executed earlier, before the bundling

> process takes place. Existing 'beforeTask' and 'afterTask' configuration of

Reviewer: ```suggestion
process takes place. Any existing 'beforeTask' or 'afterTask' configuration of
```

| **ADDED:** minify | {: .sap-icon-circle-task } | {: .sap-icon-circle-task } |  |
| generateLibraryManifest |  | {: .sap-icon-circle-task-2 } |  |
| generateLibraryPreload |  | {: .sap-icon-circle-task-2 } |  |
| generateComponentPreload | {: .sap-icon-circle-task-2 } | {: .sap-icon-circle-task }^2^ |  |
| generateStandaloneAppBundle | {: .sap-icon-circle-task }^3^ |  |  |
| transformBootstrapHtml | {: .sap-icon-circle-task }^3^ |  |  |
| generateBundle | {: .sap-icon-circle-task }^4^ | {: .sap-icon-circle-task }^4^ |  |
| buildThemes |  | {: .sap-icon-circle-task-2 } | {: .sap-icon-circle-task-2 } |
| generateThemeDesignerResources |  | {: .sap-icon-circle-task } | {: .sap-icon-circle-task } |
| **REMOVED:** ~~createDebugFiles~~ |  |  |  |
| **REMOVED:** ~~uglify~~ |  |  |  |
| generateVersionInfo | {: .sap-icon-circle-task-2 } |  |  |
| generateCachebusterInfo | {: .sap-icon-circle-task } |  |  |
| generateApiIndex | {: .sap-icon-circle-task }^1^ |  |  |
| generateResourcesJson | {: .sap-icon-circle-task } | {: .sap-icon-circle-task } | {: .sap-icon-circle-task } |

*Enabled by default*
{: .sap-icon-circle-task-2-before }

> *Disabled by default. Can be activated by certain build modes, project configuration or by using the `--include-task` [CLI parameter](../pages/CLI.md#ui5-build). See footnotes where given* 

Reviewer: ```suggestion
*Disabled by default. Can be activated by certain build modes, the project configuration, or by using the `--include-task` [CLI parameter](../pages/CLI.md#ui5-build). See footnotes where given* 
```
